library/core/src/iter/mod.rs

```diff
@@ -381,88 +382,126 @@ macro_rules! impl_fold_via_try_fold {
 pub use self::adapters::TrustedRandomAccessNoCoerce;
 #[unstable(feature = "iter_chain", reason = "recently added", issue = "125964")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::adapters::chain;
+#[cfg(not(feature = "ferrocene_certified"))]
 pub(crate) use self::adapters::try_process;
 #[stable(feature = "iter_zip", since = "1.59.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::adapters::zip;
 #[stable(feature = "rust1", since = "1.0.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::adapters::{
     Chain, Cycle, Enumerate, Filter, FilterMap, FlatMap, Fuse, Inspect, Map, Peekable, Rev, Scan,
     Skip, SkipWhile, Take, TakeWhile, Zip,
 };
 #[unstable(feature = "iter_intersperse", reason = "recently added", issue = "79524")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::adapters::{Intersperse, IntersperseWith};
 #[unstable(
     feature = "step_trait",
     reason = "likely to be replaced by finer-grained traits",
     issue = "42168"
 )]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::range::Step;
 #[unstable(feature = "iter_macro", issue = "none", reason = "generators are unstable")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::sources::iter;
 #[stable(feature = "iter_empty", since = "1.2.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::sources::{Empty, empty};
 #[unstable(
     feature = "iter_from_coroutine",
     issue = "43122",
     reason = "coroutines are unstable"
 )]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::sources::{FromCoroutine, from_coroutine};
 #[stable(feature = "iter_from_fn", since = "1.34.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::sources::{FromFn, from_fn};
 #[stable(feature = "iter_once", since = "1.2.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::sources::{Once, once};
 #[stable(feature = "iter_once_with", since = "1.43.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::sources::{OnceWith, once_with};
 #[stable(feature = "rust1", since = "1.0.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::sources::{Repeat, repeat};
 #[stable(feature = "iter_repeat_n", since = "1.82.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::sources::{RepeatN, repeat_n};
 #[stable(feature = "iterator_repeat_with", since = "1.28.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::sources::{RepeatWith, repeat_with};
 #[stable(feature = "iter_successors", since = "1.34.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::sources::{Successors, successors};
 #[stable(feature = "fused", since = "1.26.0")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::traits::FusedIterator;
 #[unstable(issue = "none", feature = "inplace_iteration")]
+#[cfg(not(feature = "ferrocene_certified"))]
 pub use self::traits::InPlaceIterable;
-#[stable(feature = "rust1", since = "1.0.0")]
-pub use self::traits::Iterator;
```

Comment:
Don't delete. Without it `./x build library/core` fails.

```suggestion
#[stable(feature = "rust1", since = "1.0.0")]
#[cfg(not(feature = "ferrocene_certified"))]
pub use self::traits::Iterator;
```